pkg/clustermesh/operator/remote_cluster.go

```diff
@@ -62,6 +65,19 @@ func (rc *remoteCluster) Run(ctx context.Context, backend kvstore.BackendOperati
 		})
 	}
 
+	if rc.clusterMeshEnableMCSAPI && config.Capabilities.ServiceExportsEnabled != nil {
+		mgr.Register(adapter(mcsapitypes.ServiceExportStorePrefix), func(ctx context.Context) {
+			rc.remoteServiceExports.Watch(ctx, backend, path.Join(adapter(mcsapitypes.ServiceExportStorePrefix), rc.name))
+		})
+	} else {
+		// Drain the remote service exports in case the remote cluster no longer supports them
+		rc.remoteServiceExports.Drain()
+		// Also mimic that service exports are synced so that consumer may
+		// use remote service exports even if service exports are disabled or
+		// not supported by the remote cluster
+		rc.remoteServiceExports.Synced()
```

Comment:
Hmm, this doesn't look correct, being a getter. You probably wanted to call `rc.synced.serviceExports.Stop()`.

Reply:
Ahh yes not the right method indeed! :sweat_smile: